app/utils/exceptions.py

```diff
@@ -48,4 +48,19 @@ def __init__(self,key):
 
 class CompanyAlreadyExistsException(HTTPException):
     def __init__(self, company_name: str):
-        super().__init__(status_code=400, detail=f"Company with the name '{company_name}' already exists")
\ No newline at end of file
+        super().__init__(status_code=400, detail=f"Company with the name '{company_name}' already exists")
+
+
+class NoSuchMemberException(HTTPException):
```

Comment:
i saied u before, its very big error to use HTTPException in services. Just inherit from base Exception and try to catch this error in ur route and IN ROUTE u could raise HTTPException